### PR TITLE
win,tty: Support consoles with non-default colors

### PR DESCRIPTION
**Pull request by mcnameej**
Fix for #424. Use the console text color set by the user rather than assuming the Windows default (white foreground on black background).

How it looks before the fix...

screen shot - blue background

How it looks after the fix...

tty fix screenshot

### CODE REVIEW

**Comment by saghul** [comment]
You have some long lines I think, other than that LGTM. @piscisaureus, care to take a quick look?

**Comment by piscisaureus** [comment]
Generally speaking, I'm in favor of this change. However your patch itself is a bit all over the place, I'll comment inline.

File: src/win/tty.c [hunk 1]
```diff
@@ -138,6 +140,10 @@ int uv_tty_init(uv_loop_t* loop, uv_tty_t* tty, uv_file fd, int readable) {
     /* shared between all uv_tty_t handles. */
     EnterCriticalSection(&uv_tty_output_lock);
 
+    /* Remember the original console text attributes. */
+    if (uv_tty_default_text_attributes == 0)
+      uv_tty_default_text_attributes = screen_buffer_info.wAttributes;
```
Comment (piscisaureus) [inline src/win/tty.c]: Are you sure `screen_buffer_info.wAttributes` is never 0 ?

File: src/win/tty.c [hunk 2]
```diff
@@ -1176,13 +1185,35 @@ static int uv_tty_set_style(uv_tty_t* handle, DWORD* error) {
   char fg_bright = -1, bg_bright = -1;
   char inverse = -1;
 
+  char default_fg_color = 7, default_bg_color = 0;
+  char default_fg_bright = 0, default_bg_bright = 0;
+  char default_inverse = 0;
+
+  if (uv_tty_default_text_attributes != 0) {
```
Comment (piscisaureus) [inline src/win/tty.c]: I don't see why you would parse it every time. Can you put all this info in a static variable, and initialize them once in `uv_tty_init()` (or, even better, a new function called `uv__tty_capture_initial_style`, and call it from `uv_tty_init()`)

File: src/win/tty.c (hunk 1 above)
Comment (mcnameej) [inline src/win/tty.c]: screen_buffer_info.wAttributes **can** be zero, but as a a practical matter, it never **will** be.  Zero would mean black text on a black background, and nobody sets a console to that.  If for some reason a console was set this way, nothing bad would happen.  tty would just fall back to it's prior behavior, and assume Windows default colors.  I didn't see a reason to create a separate "uv_tty_default_text_attributes_is_valid" boolean.

File: src/win/tty.c (hunk 2 above)
Comment (mcnameej) [inline src/win/tty.c]: If y'all don't mind a new function and static data, I'm certainly OK doing it that way.

I was trying to keep the changes localized so the patch would be easier to review.  I performed minimally invasive surgery on a code base where I'm not familiar with all the local customs and traditions. :-)

**Comment by piscisaureus** [comment]
Thanks John! I landed your patch in 517ade8, with a few minimal modifications.

**Pull request closed without merging**